### PR TITLE
New warning for %between%

### LINKED ISSUE #3014: potential issue with the between operator
The between operator seems to not operate as intended:
```
d <- data.table(A=1:10)
d[,B:=3]
d[A %between% c(B,B+1)]  ##  incorrect result since default setting includes bounds 
#    A B
# 1: 3 3
d[A %between% c(3,3+1)]  ## correct result when using same notation with hard-coded bounds
#    A B
# 1: 3 3
# 2: 4 3
d[between(A,B,B+1)]      ## correct result when using different notation
#    A B
# 1: 3 3
# 2: 4 3
```

My R session and data.table version:
```
R version 3.4.3 (2017-11-30) -- "Kite-Eating Tree"
Platform: x86_64-redhat-linux-gnu (64-bit)
> library(data.table)
data.table 1.11.2
```

Also tried:
```
 R version 3.5.1 (2018-07-02) -- "Feather Spray"
 Platform: x86_64-w64-mingw32/x64 (64-bit)
> library(data.table)
data.table 1.11.5 IN DEVELOPMENT built 2018-08-16 23:02:55 UTC; root  Latest news: r-datatable.com
```

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #3014

### CODE REVIEW

File: R/between.R [hunk 1]
```diff
@@ -12,7 +12,18 @@ between <- function(x,lower,upper,incbounds=TRUE) {
 }
 
 # %between% is vectorised, #534.
-"%between%" <- function(x, y) between(x, y[[1L]], y[[2L]], incbounds=TRUE)
+"%between%" <- function(x, y) {
+  if (length(y) > 2L) {
+    ysub = substitute(y)
+    warning("RHS has length() greater than 2. ",
```
Comment (HughParsonage) [inline R/between.R]: à la `if (1:5 > 1) "a"` we should indicate that part of the input is being ignored: "RHS has length > 2 and only the first and second elements will be used".

Comment (MichaelChirico) [reply R/between.R]: I thought to include that, but didn't want an overlong message; you don't think that's strongly implied by the warning here?

Maybe:?

> RHS has length() greater than 2 and the rest will be ignored. [...]

Comment (HughParsonage) [reply R/between.R]: I think a problem can arise when the user says `y = c(B, B + 1)`, 'knowing' that `length(y) > 2` -- duh! It's vectorized; the warning is nothing to worry about. 

On second thoughts, maybe this should be an error. Documentation is pretty clear that `y` must be length-2, it was just never enforced. I'm trying to think of a case where only a warning is appropriate.

The suggestion (`Perhaps...`) should obviously stay: that's imo the most valuable part of this PR.

Comment (MichaelChirico) [reply R/between.R]: I also was struggling to think of a use case that's not subsumed by just running `[1:2]` on whatever your input is (and even the `[1:2]` cases i couldn't think of a _real_ use case)

Comment (MichaelChirico) [reply R/between.R]: @HughParsonage updated

**Pull request merged**